### PR TITLE
[Backport stable/8.3] Add migration for restoring missing backoff jobs

### PR DESCRIPTION
**Pull request by nicpuppa**
# Description
Backport of #14329 to `stable/8.3`.

### CODE REVIEW

**Review by koevskinikola — APPROVED** [review]
LGTM :+1: , just a small nitpick

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 1]
```diff
@@ -469,4 +486,22 @@ private void removeJobBackoff(final long job, final long backoff) {
...
+    statesJobColumnFamily.forEach(
+        (key, value) -> {
+          if ((State.FAILED).equals(value.getState())) {
+            failedJobKeys.add(key.inner().getValue());
+          }
+        });
+    return failedJobKeys;
+  }
+
+  private Set<Long> getBackoffJobKey() {
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:501]: Should we change the name also in the main branch ? 😄

**Comment by nicpuppa** [comment]
bors merge

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (koevskinikola) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java]: yup, I thought I already gave the feedback there.

**Comment by nicpuppa** [comment]
bors retry

**Pull request merged**